site/src/sphinx/client-decorator.rst

```diff
@@ -67,6 +67,113 @@ If your decorator is expected to be reusable, it is recommended to define a new
+    |           |  res  | decorator |  res  | decorator |  res  | Netwokring Layer |    res    |              |
+    |           |<------|           |<------|           |<------|                  |<----------|              |
+    +-----------+       +-----------+       +-----------+       +------------------+           +--------------+
+    @endditaa
+
+If the client is a Thrift client and RPC decorators are inserted, HTTP decorators and RPC decorators are
+separately grouped and executed in reverse order of the insertion:
+
+.. code-block:: java
+
+    ClientBuilder cb = Clients.builder(...);
+
+    // #2 HTTP decorator.
+    cb.decorator((delegate, ctx, httpReq) -> {
+        System.err.println("Fourthly, executed.");
+        ...
+    });
+
+    // #2 RPC decorator.
+    cb.rpcDecorator((delegate, ctx, rpcReq) -> {
+        System.err.println("Secondly, executed.");
+        ...
+    });
+
+    // #1 HTTP decorator.
+    cb.decorator((delegate, ctx, httpReq) -> {
+        System.err.println("Thirdly, executed.");
+        ...
+    });
+
+    // #1 RPC decorator.
+    cb.rpcDecorator((delegate, ctx, rpcReq) -> {
+        System.err.println("Firstly, executed.");
+        ...
+    });
+
+An RPC request is converted into an HTTP request before it's sent to a server.
+So RPC decorators are inserted before the RPC request is converted and HTTP decorators are inserted after the
+request is converted into the HTTP request. The following diagram describes it:
+
+.. uml::
+
+    @startditaa
+    +--------+ req +-----------+ req +-----------+                     +-----------+ req +-----------+
+    |        |---->|           |---->|           |---> RPC to HTTP --->|           |---->|           |----=->
+    | Thrift |     | #1 RPC    |     | #2 RPC    |                     | #1 HTTP   |     | #2 HTTP   |
+    | Client | res | decorator | res | decorator |                     | decorator | res | decorator |
+    |        |<----|           |<----|           |<--- HTTP to RPC <---|           |<----|           |<-=----
+    +--------+     +-----------+     +-----------+                     +-----------+     +-----------+
+    @endditaa
+
+If the decorator modifies the response (e.g. :api:`DecodingClient`) or spawns more requests
+(e.g. :api:`RetryingClient`), the consequence is different depending on the order of the decorators.
+Let's look at the following example that :api:`DecodingClient` and :api:`ContentPreviewingClient`
+are used together:
+
+.. code-block:: java
+
+    import com.linecorp.armeria.client.encoding.DecodingClient;
+    import com.linecorp.armeria.client.logging.ContentPreviewingClient;
+
+    ClientBuilder cb = Clients.builder(...);
+    // Don't do this. ContentPreviewingClient should be inserted after DecodingClient.
+    cb.decorator(ContentPreviewingClient.newDecorator(1000));
+    cb.decorator(DecodingClient.newDecorator());
```

Comment:
Why don't we show the right example over the wrong one? Because many users might want to copy and paste.

```suggestion
    cb.decorator(DecodingClient.newDecorator());
    // ContentPreviewingClient should be inserted after DecodingClient.
    cb.decorator(ContentPreviewingClient.newDecorator(1000));
```

Reply:
Fixed. 😉